### PR TITLE
[WIP] Network node

### PR DESCRIPTION
**Pull request by milosgajdos83**
This PR adds a `network` command which allows to run `micro` network node.

### CODE REVIEW

File: network/network.go [hunk 1]
```diff
@@ -1 +1,158 @@
...
+	"github.com/micro/go-micro/router"
+	"github.com/micro/go-micro/tunnel"
+	"github.com/micro/go-micro/util/log"
+)
+
+var (
+	// Name of the network service
+	Name = "go.micro.network"
+	// Address is the network address
+	Address = ":8084"
```
Comment (asim) [inline network/network.go]: API 8080
Proxy 8081
Web 8082
Tunnel 8083
Router 8084
Network 8085

File: network/network.go [hunk 2]
```diff
@@ -1 +1,158 @@
...
+	// Tunnel is the name of the network tunnel
+	Tunnel = "tun:0"
```
Comment (asim) [inline network/network.go]: Unneeded

File: network/network.go [hunk 3]
```diff
@@ -1 +1,158 @@
...
+		p.Init(ctx)
+	}
+
+	if len(ctx.GlobalString("server_name")) > 0 {
+		Name = ctx.GlobalString("server_name")
+	}
+	if len(ctx.String("address")) > 0 {
+		Address = ctx.String("address")
+	}
+	if len(ctx.String("tunnel_id")) > 0 {
```
Comment (asim) [inline network/network.go]: We don't need a tunnel id since we don't use endpoints

File: network/network.go [hunk 4]
```diff
@@ -1 +1,158 @@
...
+		Tunnel = ctx.String("tunnel_id")
+		// We need host:port for the Endpoint value in the proxy
+		parts := strings.Split(Tunnel, ":")
+		if len(parts) == 1 {
+			Tunnel = Tunnel + ":0"
+		}
+	}
+	var nodes []string
+	if len(ctx.String("server")) > 0 {
```
Comment (asim) [inline network/network.go]: Should use `nodes` here as in network nodes

File: network/network.go [hunk 5]
```diff
@@ -1 +1,158 @@
...
+	case "dns":
+		res = &dns.Resolver{}
+	case "http":
+		res = &http.Resolver{}
+	case "registry":
+		res = &registry.Resolver{}
+	}
+
+	// create a tunnel
+	tun := tunnel.NewTunnel(
```
Comment (asim) [inline network/network.go]: We should pass Address and Nodes to Network and allow it to deal with creating the tunnel or calling tunnel.Init

Comment (milosgajdos83) [reply network/network.go]: We dont seem to creat tunnel in network at the moment. We said we would pass created tunnel into its constructor?

**Pull request merged**